File: programmers/기능개발.py
"""
코딩테스트 연습 > 스택/큐
"""


def solution(progresses, speeds):
    stack = []
    res = []

    for i in range(len(progresses)):
        p = progresses[i]
        s = speeds[i]

        remain = (100-p)//s
        if (100-p)%s != 0:
            remain += 1

        if len(stack) == 0 or remain <= stack[0]:
            stack.append(remain)
        else:
            res.append(len(stack))
            stack = [remain]

    if len(stack) != 0:
        res.append(len(stack))

    return res

File: programmers/기능개발.kt
class Solution {
    fun solution(progresses: IntArray, speeds: IntArray): IntArray {
        var remainDays = progresses.mapIndexed { i, v ->
            var remainProgress = 100-v
            var res = remainProgress / speeds[i]
            if (remainProgress % speeds[i] > 0) {
                res += 1
            }

            res
        }

        var res = mutableListOf<Int>()
        var flag = 0
        remainDays.forEach {
            if (it > flag) {
                res.add(1)
                flag = it
            } else {
                res[res.size-1] += 1
            }
        }

        return res.toIntArray()
    }
}
